lib/spack/spack/test/cmd/fetch.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import os
+from six import StringIO
+
+import pytest
+
+import llnl.util.filesystem as fs
+
+import spack.hash_types as ht
+import spack.modules
+import spack.environment as ev
+
+from spack.cmd.env import _env_create
+from spack.spec import Spec
+from spack.main import SpackCommand, SpackCommandError
+from spack.stage import stage_prefix
+
+from spack.util.mock_package import MockPackageMultiRepo
+import spack.util.spack_json as sjson
+from spack.util.path import substitute_path_variables
+
+
+# everything here uses the mock_env_path
+pytestmark = pytest.mark.usefixtures(
+    'mutable_mock_env_path', 'config', 'mutable_mock_repo')
+
+
+@pytest.mark.disable_clean_stage_check
+def test_fetch_in_env(tmpdir, mock_archive, mock_stage, mock_fetch, install_mockery):
+    SpackCommand('env')('create', 'test')
+    with ev.read('test'):
+        SpackCommand('add')('python')
+        with pytest.raises(SpackCommandError):
+            SpackCommand('fetch')()
```

Comment:
Shouldn't `spack fetch` in an environment download all of the packages in that environment? Why does it raise an error?

Reply:
Because at this point we don't know which packages are in the environment at all yet (didn't do concretize yet)